inc/composer/class-docker-compose-generator.php

```diff
@@ -624,6 +625,44 @@ public function get_array() : array {
 				'socket' => null,
 			],
 		];
+
+		// Handle docker-sync volume according to args.
+		if ( $this->args['docker-sync'] ?? false ) {
+			$config['volumes'][ 'app-sync-' . $this->project_name ] = [
+				'external' => true,
+			];
+		}
+
+		// Handle mutagen volume according to args.
+		if ( $this->args['mutagen'] ?? false ) {
+			$config['volumes']['app'] = null;
+			$config['x-mutagen'] = [
+				'sync' => [
+					'app' => [
+						'alpha' => $this->root_dir,
+						'beta' => 'volume://app',
+						'configurationBeta' => [
+							'permissions' => [
+								'defaultOwner' => 'id:82',
+								'defaultGroup' => 'id:82',
+								'defaultFileMode' => '0664',
+								'defaultDirectoryMode' => '0775',
+							],
+						],
+						'mode' => 'two-way-resolved',
+						'ignore' => [
+							'paths' => [
+								'.DS_Store',
+								'.git',
+								'node_modules',
```

Comment:
This technically differs from what we have now, which I think _could_ lead to some unexpected behavior.

Reply:
Hm, it's incredibly unlikely. Git files and node_modules are only interacted with on the host machine, both node and git are not installed on any of the containers were sharing files with.

This is really one of the main advantages and the reason for doing this in the first place. Take the product-dev repo for example. If I'm working on say analytics and smart media and have `node_modules` installed for both I could have over 200k files in my project, doing this means it only needs to worry about sharing something closer to 40k files so it's much more efficient.

Reply:
Well, in the current implementation, the amount of files you are sharing doesn't impact anything, because it's a filesystem wrapper. Unless the guest machine tries to read them, there is no overhead to move files. That might not be the same with your two supported methods of syncing files however.

I wonder if we could at least support a config option for it -- I know written code that can interact with node_modules (with things like v8js bridges, manifests etc).